Sources/CombineWamp/PeerWorkflow/Connection/WampSession.swift

```diff
@@ -77,7 +87,21 @@ public class WampSession: Cancellable {
 }
 
 extension WampSession {
+
+    public var isConnected: Bool {
+        connectedQueue.sync {
```

Comment:
Shouldn't this also use the barrier? If I understand correctly, it doesn't matter that sync blocks others, because sync is already blocking by definition, however without the barrier I understand that it will not wait for the async with barriers (maybe I'm wrong). If this is the case, there's a risk of isConnected returning false even with a connected=true scheduled with barrier to be updated, and therefore, if you immediately ask the status you would have the wrong value.

Again, I might be wrong. Tests would be great.

https://developer.apple.com/documentation/dispatch/1452917-dispatch_barrier_sync